presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -46,7 +38,7 @@
 import static java.util.Objects.requireNonNull;
 
 public class CachingJdbcClient
-        implements JdbcClient
+        extends ForwardingJdbcClient
```

Comment:
I would rather delegate explicitly. And ensure it delegates everything with `testEverythingImplemented`.
This will ensure that if we add a new method (like `schemaexists(ConnectorSesson, String)` overload), this class will get updated properly.

So IMO this class should be impl the same way the StatisticsAware class is -- it has the same design principles of overriding everything except handed-picked exceptions (if any)

Reply:
This was done this way in purpose. Please undo this commit.

Reply:
It's not possible to delegate `schemaExists` as the default implementation in the delegated would call `getSchemaNames()` on itself, thus bypassing the cache. We'd need some way to specify exceptions for `assertAllMethodsOverridden()`. But doing that and being explicit does seem better.